django/db/backends/oracle/operations.py

```diff
@@ -676,6 +676,25 @@ def bulk_insert_sql(self, fields, placeholder_rows):
             for field in fields
             if field
         ]
+        if (
+            self.connection.features.supports_insert_multiple_rows
+            # A workaround with UNION of SELECTs is required for models without
+            # any fields.
+            and field_placeholders
+        ):
+            placeholder_rows_sql = []
+            for row in placeholder_rows:
+                placeholders_row = (
+                    field_placeholders[i] % placeholder
+                    for i, placeholder in enumerate(row)
+                )
+                placeholder_rows_sql.append(
+                    ", ".join([row for row in placeholders_row])
+                )
+            values_sql = ", ".join([f"({sql})" for sql in placeholder_rows_sql])
+            return f"VALUES {values_sql}"
```

Comment:
With the default approach being hoisted into the backend, this can then be simplified to the following:

```suggestion
            placeholder_rows = (
                field_placeholders[i] % placeholder
                for row in placeholder_rows
                for i, placeholder in enumerate(row)
            )
            return super().bulk_insert_sql(fields, placeholder_rows)
```

From the original version below, there is an `if fields[i]:` check... I'm unclear as to why that isn't necessary here? If it is necessary still, then this would become:

```suggestion
            placeholder_rows = (
                (field_placeholders[i] % placeholder if fields[i] else placeholder)
                for row in placeholder_rows
                for i, placeholder in enumerate(row)
            )
            return super().bulk_insert_sql(fields, placeholder_rows)
```

Otherwise, in combination with Simon's suggestion we can simplify to:

```suggestion
            placeholder_rows = (
                field_placeholder % placeholder
                for row in placeholder_rows
                for field_placeholder, placeholder in zip(field_placeholders, row)
            )
            return super().bulk_insert_sql(fields, placeholder_rows)
```

Reply:
Your suggestion won't work as `placeholder_rows` must be a tuple of tuples of placeholders and not flat list of placeholders. Nevertheless, I added `zip()` and reused based implementation.

Reply:
Yes, sorry - it'd be more like this:

```python
            placeholder_rows = (
                (
                    field_placeholder % placeholder
                    for field_placeholder, placeholder in zip(field_placeholders, row)
                ) for row in placeholder_rows
            )
            return super().bulk_insert_sql(fields, placeholder_rows)
```

But never mind, it's already much nicer!